telemetry_device/TelemetryDevice.cs

```diff
@@ -0,0 +1,80 @@
+﻿using simulator_main.icd;
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+using telemetry_device_main.decryptor;
+using telemetry_device_main.icds;
+namespace telemetry_device
+{
+    class TelemetryDevice
+    {
+        const int PORT = 50000;
+        TcpClient simulator;
+        NetworkStream stream;
+        private Dictionary<IcdTypes, (Type, string)> IcdDictionary;
+        const string FILE_TYPE = ".json";
+        const string REPO_PATH = "../../../icd_repo/";
+        public TelemetryDevice()
+        {
+            this.IcdDictionary = new Dictionary<IcdTypes, (Type, string)>();
+            IcdDictionary.Add(IcdTypes.FlightBoxDown, (typeof(FlightBoxIcd), "FlightBoxUpIcd"));
+            IcdDictionary.Add(IcdTypes.FlightBoxUp, (typeof(FlightBoxIcd), "FlightboxUpIcd"));
+            IcdDictionary.Add(IcdTypes.FiberBoxDown, (typeof(FiberBoxIcd), "FiberBoxDownIcd"));
+            IcdDictionary.Add(IcdTypes.FiberBoxUp, (typeof(FiberBoxIcd), "FiberBoxUpIcd"));
+        }
+        public async Task ConnectAsync()
+        {
+            IPAddress ipAddr = new IPAddress(new byte[] { 127, 0, 0, 1 });
+            TcpListener socket = new TcpListener(ipAddr, PORT);
+            socket.Start(1);
+            simulator = await socket.AcceptTcpClientAsync();
+            stream = simulator.GetStream();
+        }
+        public async Task RunAsync()
+        {
+            await ConnectAsync();
+
+            Task.Run(() => { ListenForPackets(); });
+            await Task.Delay(-1);
```

Comment:
why -1?

Reply:
without it the whole program closes after connecting to the simulator, it is the same as Task.WaitAll

Reply:
Ok, add comment here